src/web/client/test/integration/errorHandler.test.ts

```diff
@@ -0,0 +1,634 @@
+
+    it("checkMandatoryQueryParameters__shouldReturnTrueWhenAllParameterIsPassedAndValid", () => {
+        //Act
+        const appName = "portal";
+        const queryParamsMap = new Map<string, string>([
+            [queryParameters.ORG_URL, "orgUrl"],
+            [queryParameters.DATA_SOURCE, "SQL"],
+            [schemaKey.SCHEMA_VERSION, "1.0.0.0"],
+            [
+                queryParameters.WEBSITE_ID,
+                "ed9a6c19-5ab6-4f67-8c35-2423cff958c4",
+            ],
+        ]);
+        //Action
+        const result = checkMandatoryQueryParameters(appName, queryParamsMap);
+        //Assert
+        expect(result).true;
+    });
+
+    it("checkMandatoryQueryParameters_shouldReturnFalseWhenORGURLIsBlank", () => {
+        //Act
+        const _mockShowErrorMessage = spy(vscode.window, "showErrorMessage");
+        const _mockSendErrorTelemetry = spy(
+            powerPlatformExtensionContext.telemetry,
+            "sendErrorTelemetry"
+        );
+
+        const appName = "portal";
+        const queryParamsMap = new Map<string, string>([
+            [queryParameters.ORG_URL, ""],
+            [queryParameters.DATA_SOURCE, "SQL"],
+            [schemaKey.SCHEMA_VERSION, "1.0.0.0"],
+            [
+                queryParameters.WEBSITE_ID,
+                "ed9a6c19-5ab6-4f67-8c35-2423cff958c4",
+            ],
+        ]);
+        //Action
+        const result = checkMandatoryQueryParameters(appName, queryParamsMap);
+        //Assert
+        expect(result).false;
+        const detailMessage = localize(
+            "microsoft-powerapps-portals.webExtension.parameter.desc",
+            "Check the URL and verify the parameters are correct"
+        );
+
+        const errorString = localize(
+            "microsoft-powerapps-portals.webExtension.parameter.error",
+            "There was a problem opening the workspace"
+        );
+        const options = { detail: detailMessage, modal: true };
+
+        assert.calledOnceWithExactly(
+            _mockShowErrorMessage,
+            errorString,
+            options
+        );
+        assert.calledOnceWithExactly(
+            _mockSendErrorTelemetry,
+            telemetryEventNames.WEB_EXTENSION_MANDATORY_QUERY_PARAMETERS_MISSING
+        );
+    });
+
+    it("checkMandatoryQueryParameters_shouldReturnFalseWhenDATASOURCEIsBlank", () => {
+        //Act
+        const _mockShowErrorMessage = spy(vscode.window, "showErrorMessage");
+        const _mockSendErrorTelemetry = spy(
+            powerPlatformExtensionContext.telemetry,
+            "sendErrorTelemetry"
+        );
+        const appName = "portal";
+        const queryParamsMap = new Map<string, string>([
+            [queryParameters.ORG_URL, "ORG_URL"],
+            [queryParameters.DATA_SOURCE, ""],
+            [schemaKey.SCHEMA_VERSION, "1.0.0.0"],
+            [
+                queryParameters.WEBSITE_ID,
+                "ed9a6c19-5ab6-4f67-8c35-2423cff958c4",
+            ],
+        ]);
```

Comment:
I see lot of common code when orgURL/ dataSource /schemaName /websiteId is null.
Keep a common method and call it.

Reply:
But we are sending different different data in orgURL/ dataSource /schemaName /websiteId